### PR TITLE
Get queries from the new cactvs SMARTS extensions to pickle correctly

### PR DESCRIPTION
**Pull request by greglandrum**
Fixes a problem pickling/unpickling range queries.
Fixes the unpickling of the new cactvs extensions added in #1704

### CODE REVIEW

File: Code/GraphMol/MolPickler.cpp [hunk 1]
```diff
@@ -408,6 +428,9 @@ Query<int, T const *, true> *buildBaseQuery(std::istream &ss, T const *owner,
       static_cast<RangeQuery<int, T const *, true> *>(res)->setUpper(val);
       streamRead(ss, val, version);
       static_cast<RangeQuery<int, T const *, true> *>(res)->setTol(val);
+      streamRead(ss, cval, version);
+      static_cast<RangeQuery<int, T const *, true> *>(res)->setEndsOpen(
+          cval & (1 << 1), cval & 1);
```
Comment (bp-kelley) [inline Code/GraphMol/MolPickler.cpp]: Why use the shift op here?  It might be cleaner as (cval & 0x2).

Comment (bp-kelley) [reply Code/GraphMol/MolPickler.cpp]: I see, it's a mirror of the pickler.

Comment (bp-kelley) [reply Code/GraphMol/MolPickler.cpp]: I see, it's a mirror of the pickler
```
    ends = 0 | (rdcast<int>(lowerOpen) << 1) | rdcast<int>(upperOpen);
```

Still, might be easier as the following:

```
const unsigned int LowerOpen = (1<<1);
const unsigned int UpperOpen = 1;
...
static_cast<RangeQuery<int, T const *, true> *>(res)->setEndsOpen(cval & LowerOpen, cval & UpperOpen);
          
```

**Comment by greglandrum** [comment]
Agreed, that makes it a bit easier to read. Updated.
I'll merge when the travis build finishes.

**Pull request merged**